py-polars/polars/dataframe/frame.py

```diff
@@ -3834,6 +3835,32 @@ def unpack_table_name(name: str) -> tuple[str | None, str | None, str]:
             msg = f"unrecognised connection type {connection!r}"
             raise TypeError(msg)
 
+    def write_iceberg(
+        self,
+        table: pyiceberg.table.Table,
+        mode: Literal["append", "overwrite"],
+    ) -> None:
+        """
+        Write DataFrame to an Iceberg table.
+
+        Parameters
+        ----------
+        table
+            The pyiceberg.table.Table object representing an Iceberg table.
+        mode : {'append', 'overwrite'}
+            How to handle existing data.
+
+            - If 'append', will add new data.
+            - If 'overwrite', will replace table with new data.
+
+        """
+        data = self.to_arrow()
+
+        if mode == "append":
+            table.append(data)
+        else:
+            table.overwrite(data)
```

Comment:
This logic is so simple that this does not warrant its own method, in my opinion.

Reply:
The main goal is to implement the top-level `write_iceberg` function for the dataframe API. The actual implementation code can be moved somewhere else

Reply:
That is true. Pyiceberg is well integrated with arrow. With an arrow dataframe and an Iceberg table, one can just invoke the pyiceberg write functions `.append`/`.overwrite`. 

Given the above, is there still value in implementing a simple `write_iceberg` method? 

Looking at the [`write_delta`](https://github.com/pola-rs/polars/blob/f803053fb73abda45bd11ad25fc0a6e642e1038e/py-polars/polars/dataframe/frame.py#L3859-L4059) method, aside from the `merge` functionality, the function just pass data as arrow into the `write_deltalake` function.

Reply:
> If there is some complex logic required to set up the iceberg table

There's a scenario where a user might want to write an Iceberg table to a location in blob store. In such case, the `write_iceberg` function can take care of creating an in-memory catalog and iceberg table object before writing. 

That was the initial version of this PR 
https://github.com/pola-rs/polars/pull/15018/commits/08010124c3a67fdd759df83006a7ffef758f4572

Reply:
Seems like a write_iceberg would be worth it just by virtue of having full support from polars / parity with delta.